File: fe/lib/models/itinerary.dart
class Itinerary {
  final String id;
  final String itineraryId;
  final String userId;
  final String destination;
  final TravelProfile travelProfile;
  final List<ItineraryDay> schedule;
  final bool isSaved;
  final bool aiGenerated;
  final DateTime createdAt;

  Itinerary({
    required this.id,
    required this.itineraryId,
    required this.userId,
    required this.destination,
    required this.travelProfile,
    required this.schedule,
    this.isSaved = false,
    this.aiGenerated = false,
    required this.createdAt,
  });

  factory Itinerary.fromJson(Map<String, dynamic> json) {
    return Itinerary(
      id: json['_id'] ?? '',
      itineraryId: json['itineraryId'] ?? '',
      userId: json['userId'] ?? '',
      destination: json['destination'] ?? '',
      travelProfile: TravelProfile.fromJson(json['travelProfile'] ?? {}),
      schedule: (json['schedule'] as List? ?? [])
          .map((day) => ItineraryDay.fromJson(day))
          .toList(),
      isSaved: json['isSaved'] ?? false,
      aiGenerated: json['aiGenerated'] ?? false,
      createdAt: DateTime.parse(
        json['createdAt'] ?? DateTime.now().toIso8601String(),
      ),
    );
  }
}

class TravelProfile {
  final int people;
  final String budget;
  final List<String> interests;

  TravelProfile({
    required this.people,
    required this.budget,
    required this.interests,
  });

  factory TravelProfile.fromJson(Map<String, dynamic> json) {
    return TravelProfile(
      people: json['people'] ?? 1,
      budget: json['budget'] ?? '',
      interests: List<String>.from(json['interests'] ?? []),
    );
  }
}

class ItineraryDay {
  final int day;
  final String title;
  final List<String> activities;

  ItineraryDay({
    required this.day,
    required this.title,
    required this.activities,
  });

  factory ItineraryDay.fromJson(Map<String, dynamic> json) {
    return ItineraryDay(
      day: json['day'] ?? 1,
      title: json['title'] ?? '',
      activities: List<String>.from(json['activities'] ?? []),
    );
  }
}

File: BE/models/Itinerary.js
const mongoose = require('mongoose');

const activitySchema = new mongoose.Schema({
  time: String,
  title: String,
  placeId: String,
  notes: String,
  estimatedDuration: Number,
  estimatedCost: Number
}, { _id: false });

const dailyScheduleSchema = new mongoose.Schema({
  day: Number,
  date: String,
  theme: String,
  weatherForecast: String,
  aiTip: String,
  activities: [activitySchema]
}, { _id: false });

const travelProfileSchema = new mongoose.Schema({
  people: Number,
  budget: String,
  interests: [String],
  travelStyle: String
}, { _id: false });

const itinerarySchema = new mongoose.Schema({
  itineraryId: {
    type: String,
    unique: true,
    required: true
  },
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  destination: {
    type: String,
    required: true
  },
  travelProfile: travelProfileSchema,
  schedule: [dailyScheduleSchema],
  isSaved: {
    type: Boolean,
    default: false
  },
  aiGenerated: {
    type: Boolean,
    default: true
  },
  queryId: String,
  totalEstimatedCost: Number,
  status: {
    type: String,
    enum: ['draft', 'planned', 'active', 'completed', 'cancelled'],
    default: 'draft'
  }
}, {
  timestamps: true
});

itinerarySchema.index({ userId: 1, createdAt: -1 });
itinerarySchema.index({ destination: 1 });

module.exports = mongoose.model('Itinerary', itinerarySchema);
